data/impl/RecordSet.js

```diff
@@ -17,52 +15,38 @@ import {Record} from '../Record';
  */
 export class RecordSet {
 
-    /** @member {BaseStore} - source store. */
     store;
-    /** @member {Map} - map of all records by id. */
-    records;
-
-    /** @member {Record[]} - lazily constructed array of Records. */
-    _list;
-    /** @member {RecordNode[]} - lazily constructed array of root RecordNodes. */
-    _tree;
-
-    /**
-     * @param {BaseStore} store
-     * @param {Map} [records]
-     */
+    records;        // Records by id
+    childrenMap;    // Children by parentId
+    _list;          // Lazy, list representation of all records.
+
     constructor(store, records = new Map()) {
-        this.records = records;
         this.store = store;
+        this.records = records;
+
+        const childrenMap = this.childrenMap = new Map();
+        records.forEach(record => {
+            const {parentId} = record,
+                children = childrenMap.get(parentId);
+            if (!children) {
+                childrenMap.set(parentId, [record]);
+            } else {
+                children.push(record);
+            }
+        });
     }
 
-    /** Total number of records contained in this RecordSet. */
     get count() {
-        return this.records.size;
+        return this.records.size();
```

Comment:
Don't think this is correct, `records` is a Map and `size` is a property of Map.